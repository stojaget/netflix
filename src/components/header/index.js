import React, {useState} from 'react';
import {Link as ReactRouterLink} from 'react-router-dom';
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Profile,
  Dropdown,
  SearchIcon,
  Search,
  SearchInput,
  PlayButton,
} from "./styles/header";

export default function Header({ bg=true, children, ...restProps }){
    return bg ? <Background {...restProps} data-testid="header-bg">{children}</Background> : children;
}
Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};
Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};
Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};
Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};
Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};
Header.Search = function HeaderSearch({searchTerm, setSearchTerm, children, ...restProps }) {
  const [searchActive, setSearchActive] = useState(false);
  return (<Search {...restProps}>
  <SearchIcon onClick={() => setSearchActive((searchActive) => !searchActive)} data-testid="search-click" >
  <img src="/images/icons/search.png" alt="search"/>
  </SearchIcon>
  <SearchInput value={searchTerm} onChange={({target}) => setSearchTerm(target.value)}
   placeholder="Search films and series" active={searchActive} data-testid="search-input">
   </SearchInput>
  </Search>);
};
Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`}/>;
};
Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};
Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};
Header.Feature = function HeaderFeature({children, ...restProps}){
    return <Feature {...restProps}>{children}</Feature>;
}
Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};
Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};
Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
      <ReactRouterLink to={to}>
      <Logo {...restProps} />
      </ReactRouterLink>

  );
};

