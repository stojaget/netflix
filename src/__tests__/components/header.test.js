import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Header } from '../../components';
jest.mock('react-router-dom');
describe('<Header />', () => {
    it('renders the basic <Header /> component without a background', () => {
const {container, getByTestId, getByText} = render(
<Header bg={false}>
<Header.Frame>
<Header.Logo src="/logo.svg" alt="Netflix"/>
<Header.ButtonLink active="true">Sign In</Header.ButtonLink>
<Header.TextLink active="false">Hello I am a link!</Header.TextLink>
</Header.Frame>
</Header>
);
 expect(getByText('Hello I am a link!')).toBeTruthy();
 expect(getByTestId('header-bg')).toBeFalsy();
          expect(container.firstChild).toMatchSnapshot();
    });
    it('renders the full <Header /> component with a background', () => {
const {container, getByTestId, getByText} = render(
 <Header src="joker1" dontShowOnSmallViewPort>
         <Header.Frame>
           <Header.Group>
             <Header.Logo  src="/images/logo.svg" alt="Netflix" />
             <Header.TextLink
             >
               Series
             </Header.TextLink>
             <Header.TextLink>
               Films
             </Header.TextLink>
           </Header.Group>
           <Header.Group>
             <Header.Search
               searchTerm="Joker"
               setSearchTerm={() => {}}
             />
             <Header.Profile>
               <Header.Picture src="/images/users/1.png" />
               <Header.Dropdown>
                 <Header.Group>
                   <Header.Picture src="/images/users/1.png" />
                   <Header.TextLink>darac</Header.TextLink>
                 </Header.Group>
                 <Header.Group>
                   <Header.TextLink onClick={() => {}}>
                     Sign out
                   </Header.TextLink>
                 </Header.Group>
               </Header.Dropdown>
             </Header.Profile>
           </Header.Group>
         </Header.Frame>

         <Header.Feature>
           <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
           <Header.Text>
             Forever alone in a crowd, failed comedian Arthur Fleck seeks
             connection as he walks the streets of Gotham City. Arthur wears two
             masks -- the one he paints for his day job as a clown, and the
             guise he projects in a futile attempt to feel like he's part of the
             world around him.
           </Header.Text>
           <Header.PlayButton>Play</Header.PlayButton>
         </Header.Feature>
       </Header>
);
 
 expect(getByTestId('search-input')).toBeTruthy();
 expect(getByTestId('search-input').value).toBe('Joker');
 
 fireEvent.change(getByTestId('search-input'), {target: {value: 'Simpsons'}});
 fireEvent.click(getByTestId('search-click'));
 expect(getByText('Series')).toBeTruthy();
 expect(getByText('Films')).toBeTruthy();
 expect(container.firstChild).toMatchSnapshot();
    });

    });