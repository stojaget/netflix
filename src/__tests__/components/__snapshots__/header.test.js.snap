// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> component with a background 1`] = `
<div
  class="header__Background-djltcn-0 lglpeb"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-1 cKeoqP"
  >
    <p
      class="header__Link-djltcn-2 dHwxJN"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the basic <Header /> component with a background 2`] = `
<div
  class="header__Background-djltcn-0 lglpeb"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-1 cKeoqP"
  >
    <p
      class="header__Link-djltcn-2 fplXiN"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the basic <Header /> component without a background 1`] = `
<div
  class="header__Background-djltcn-0 lglpeb"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-1 cKeoqP"
  >
    <p
      class="header__Link-djltcn-2 fplXiN"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the full <Header /> component with a background 1`] = `
<div
  class="header__Background-djltcn-0 dUzWEZ"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-djltcn-1 cKeoqP"
  >
    <div
      class="header__Group-djltcn-3 hAmgWk"
    >
      <p
        class="header__Link-djltcn-2 fplXiN"
      >
        Series
      </p>
      <p
        class="header__Link-djltcn-2 fplXiN"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-djltcn-3 hAmgWk"
    >
      <div
        class="header__Search-djltcn-5 kEDhSA"
      >
        <button
          class="header__SearchIcon-djltcn-6 ivOsNR"
        >
          <img
            alt="search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-djltcn-4 gZUbTh"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-djltcn-10 kVuknk"
      >
        <button
          class="header__Picture-djltcn-8 ePvxNk"
          src="/images/users//images/users/1.png.png"
        />
        <div
          class="header__Dropdown-djltcn-9 fxiggt"
        >
          <div
            class="header__Group-djltcn-3 hAmgWk"
          >
            <button
              class="header__Picture-djltcn-8 ePvxNk"
              src="/images/users//images/users/1.png.png"
            />
            <p
              class="header__Link-djltcn-2 fplXiN"
            >
              darac
            </p>
          </div>
          <div
            class="header__Group-djltcn-3 hAmgWk"
          >
            <p
              class="header__Link-djltcn-2 fplXiN"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-djltcn-1 header__Feature-djltcn-11 cKeoqP flTBnt"
  >
    <h2
      class="header__FeatureCallOut-djltcn-12 UDRwT"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-djltcn-13 gGXzIj"
    >
      Forever alone in a crowd, failed comedian Arthur Fleck seeks connection as he walks the streets of Gotham City. Arthur wears two masks -- the one he paints for his day job as a clown, and the guise he projects in a futile attempt to feel like he's part of the world around him.
    </p>
    <button
      class="header__PlayButton-djltcn-15 SxCeS"
    >
      Play
    </button>
  </div>
</div>
`;
